### PR TITLE
Adding try/catch to make sure kernel does not die with runtime exceptions

### PR DESCRIPTION
**Pull request by anpaz-msft**

### CODE REVIEW

File: src/Engines/BaseEngine.cs [hunk 1]
```diff
@@ -236,43 +236,57 @@ internal MimeBundle EncodeForDisplay(object displayable)
...
-                new Message
-                {
-                    Header = new MessageHeader
-                    {
-                        MessageType = "display_data"
-                    },
-                    Content = new DisplayDataContent
+            try
+            {
+                if (displayable == null) throw new ArgumentNullException(nameof(displayable));
```
Comment (cgranade) [inline src/Engines/BaseEngine.cs]: Should this line be outside the `try`/`catch` so that we can find `null` inputs?

Comment (anpaz-msft) [reply src/Engines/BaseEngine.cs]: Thanks for bringing this up.
I looked into the places where the code was throwing for null arguments and I believe we were being to restrictive on not allowing null arguments. In particular, Encoders should support encoding null data (simply ignore it), after all, `null` is a valid result from calling an operation.
So I changed the code in the `BaseEngine` to be a lot more liberal for `null` arguments, which makes it a lot more robust -it was crashing very easily-. 
Could you please take a new look?

File: src/Engines/BaseEngine.cs [hunk 2]
```diff
@@ -39,19 +39,17 @@ public ExecutionChannel(BaseEngine engine, Message parent)
...
 
             public void Stderr(string message)
             {
-                if (message == null) throw new ArgumentNullException(nameof(message));
+                if (message == null) return;
                 engine.WriteToStream(parent, StreamName.StandardError, message);
             }
 
             public void Stdout(string message)
             {
```
Comment (cgranade) [inline src/Engines/BaseEngine.cs]: Do we need a similar `if (message == null) return;` line like at line 47?

File: src/Engines/BaseEngine.cs [hunk 3]
```diff
@@ -202,31 +200,39 @@ public void RegisterDefaultEncoders()
...
+                            {
+                                displayData.Metadata[mimeType] = encoded.Value.Metadata;
+                            }
+                            break;
                         }
-                        break;
+                    }
+                    catch (Exception e)
+                    {
+                        this.Logger.LogWarning($"Encoder {encoder.GetType().FullName} threw an exception encoding '{displayable}' ({e.Message}).");
```
Comment (cgranade) [inline src/Engines/BaseEngine.cs]: Do we want to log the exception as well in this case?
```suggestion
                        this.Logger.LogWarning(e, $"Encoder {encoder.GetType().FullName} threw an exception encoding '{displayable}' ({e.Message}).");
```

File: src/Engines/BaseEngine.cs (hunk 2 above)
Comment (anpaz-msft) [reply src/Engines/BaseEngine.cs]: I actually did the opposite. I let it through even if message is null.

Comment (cgranade) [reply src/Engines/BaseEngine.cs]: Makes sense, mainly wanted to suggest that both `Stdout` and `Stderr` worked in the same way.

**Pull request merged**